test/config/cascadingdelete.go

```diff
@@ -0,0 +1,125 @@
+// SPDX-FileCopyrightText: 2022-present Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package config
+
+import (
+	"github.com/onosproject/onos-api/go/onos/topo"
+	gnmiutils "github.com/onosproject/onos-config/test/utils/gnmi"
+	"github.com/onosproject/onos-config/test/utils/proto"
+	gnmiapi "github.com/openconfig/gnmi/proto/gnmi"
+	"github.com/stretchr/testify/assert"
+	"testing"
+	"time"
+)
+
+// TestCascadingDelete checks that when a root leaf or an intermediate leaf is removed, all its children are removed too
+func (s *TestSuite) TestCascadingDelete(t *testing.T) {
+	const (
+		switchName      = "testswitch"
```

Comment:
You need to change these paths to use gnmi simulator model for testing

Reply:
Adjusted per our discussion